src/openrct2/ride/Vehicle.cpp

```diff
@@ -7151,114 +7160,101 @@ void Vehicle::UpdateAdditionalAnimation()
                             }();
                             int32_t directionIndex = sprite_direction >> 1;
                             auto offset = SteamParticleOffsets[typeIndex][directionIndex];
-                            SteamParticle::Create({ x + offset.x, y + offset.y, z + offset.z });
+                            SteamParticle::Create(CoordsXYZ(
+                                x + offset.x * vehicleEntry->SteamEffect.Longitudinal / SteamEffectTranslationCoefficient,
+                                y + offset.y * vehicleEntry->SteamEffect.Longitudinal / SteamEffectTranslationCoefficient,
+                                z + offset.z * vehicleEntry->SteamEffect.Vertical / SteamEffectTranslationCoefficient));
```

Comment:
I just remembered OpenLoco exists. I think backporting it could be a better way to handle this, under the assumption it looks better than this integer division.